pkg/pillar/cmd/nim/serverdns.go

```diff
@@ -0,0 +1,189 @@
+package nim
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"time"
+
+	dns "github.com/Focinfi/go-dns-resolver"
+	"github.com/lf-edge/eve/pkg/pillar/pubsub"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+)
+
+const (
+	minTTLSec       int = 30
+	maxTTLSec       int = 3600
+	extraSec        int = 10
+	etcHostFileName     = "/etc/hosts"
+	tmpHostFileName     = "/tmp/etchosts"
+	resolvFileName      = "/etc/resolv.conf"
+)
+
+// go routine for dns query to the controller
+func queryCloudDNS(ps *pubsub.PubSub) {
+	var etchosts, cloudServer []byte
+	var ttlSec int
+	var ipaddrCached string
+
+	if _, err := os.Stat(etcHostFileName); err == nil {
+		etchosts, err = ioutil.ReadFile(etcHostFileName)
+		if err == nil {
+			cloudServer, _ = ioutil.ReadFile(types.ServerFileName)
+			cloudServer = bytes.TrimSuffix(cloudServer, []byte("\n"))
+			if bytes.Contains(cloudServer, []byte(":")) {
+				serverport := bytes.Split(cloudServer, []byte(":"))
+				if len(serverport) == 2 {
+					cloudServer = serverport[0]
+				}
+			}
+		}
+	}
+
+	if len(cloudServer) == 0 {
+		log.Errorf("can't read /etc/hosts or server file")
+		return
+	}
+
+	dnsTimer := time.NewTimer(time.Duration(minTTLSec) * time.Second)
+
+	wdName := agentName + "dnsQuery"
+	stillRunning := time.NewTicker(stillRunTime)
+	ps.StillRunning(wdName, warningTime, errorTime)
+	ps.RegisterFileWatchdog(wdName)
+
+	for {
+		select {
+		case <-dnsTimer.C:
+			// base on ttl from server dns update frequency for controller IP resolve
+			// initial wait for 100 seconds to let adapters to obtain addresses
```

Comment:
Where is this 100 second value in the code?

Reply:
forgot to remove this when moving this code to nim. will fix.